SDWebImage/SDWebImageDownloader.m

```diff
@@ -308,6 +308,15 @@ - (nullable SDWebImageDownloadToken *)downloadImageWithURL:(nullable NSURL *)url
         [self.downloadQueue addOperation:operation];
     }
     UNLOCK(self.operationsLock);
+    
+    // Prioritize the operation queue priority when the current operation is pending in queue
+    if ((options & SDWebImageDownloaderHighPriority || options & SDWebImageDownloaderLowPriority) && !operation.isFinished && !operation.isExecuting) {
+        if (options & SDWebImageDownloaderHighPriority) {
+            operation.queuePriority = NSOperationQueuePriorityHigh;
+        } else if (options & SDWebImageDownloaderLowPriority) {
```

Comment:
I'm not sure that we want to check for low priority in this case. I think that we should always use the highest requested priority possible. If we request an image with high priority and then low priority, I believe we should still use high priority in this case.

Reply:
Also thank you for taking the time to create this PR!

Reply:
@jerelevine The low priority is always set by user. And since it's always writrable unless the operation been pop from the queue, I think we just use the final option is OK.

Anyway, this does not effect too much of behavior the operation was running. Just a hint to NSOperationQueue.